components/movementsensor/gpsrtkserial/gpsrtkserial.go

```diff
@@ -662,3 +752,164 @@ func (g *rtkSerial) Close(ctx context.Context) error {
 	}
 	return nil
 }
+
+// sendGGAMessage sends GGA messages to the NTRIP Caster over a TCP connection
+// to get the NTRIP steam when the mount point is a Virtual Reference Station.
+func (g *rtkSerial) sendGGAMessage() error {
+	g.ntripMu.Lock()
+	defer g.ntripMu.Unlock()
+
+	if !g.isConnected {
+		g.readerWriter = g.connectToVirtualBase()
+	}
+
+	// read from the socket until we know if a successful connection has been
+	// established.
+	for {
+		if g.readerWriter.Reader == nil || g.readerWriter.Writer == nil {
+			break
+		}
+		line, _, err := g.readerWriter.ReadLine()
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				g.logger.Debug("EOF encountered. sending GGA message again")
+				g.readerWriter = nil
+				g.isConnected = false
+				return err
+			}
+			g.logger.Error("Failed to read server response:", err)
+			return err
+		}
+
+		if strings.HasPrefix(string(line), "HTTP/1.1 ") {
+			if strings.Contains(string(line), "200 OK") {
+				g.isConnected = true
+				break
+			} else {
+				g.logger.Error("Bad HTTP response")
```

Comment:
Consider logging the response. Just logging that something went wrong will make it hard to debug problems.